plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/AlwaysExternalCustomServerEvaluationStrategy.java

```diff
@@ -56,4 +57,19 @@ public AlwaysExternalCustomServerEvaluationStrategy(
       ContainerInfo containerInfo, String internalHost) {
     return super.getExternalAddressesAndPorts(containerInfo, internalHost);
   }
+
+  @Override
+  protected ServerImpl updateServer(ServerImpl server) {
+    super.updateServer(server);
+    ServerPropertiesImpl properties = (ServerPropertiesImpl) server.getProperties();
```

Comment:
+1 to merge, but I would probably extract all the logic to a separate method with meaningful name  / javadoc e.g.`updateProtocolForInternalUrl`